Tests/MongoSwiftSyncTests/SpecTestRunner/TestOperation.swift

```diff
@@ -57,19 +75,138 @@ struct TestOperationDescription: Decodable {
             target = .collection(collection)
         case .gridfsbucket:
             throw TestError(message: "gridfs tests should be skipped")
+        case .session0:
+            guard let session0 = sessionDict["session0"] else {
+                throw TestError(message: "got session0 object but was not provided a session")
+            }
+            target = .session(session0)
+        case .session1:
+            guard let session1 = sessionDict["session1"] else {
+                throw TestError(message: "got session1 object but was not provided a session")
+            }
+            target = .session(session1)
+        case .testRunner:
+            target = .testRunner(database)
         }
 
         do {
-            let result = try self.operation.execute(on: target, session: session)
+            let result = try self.operation.execute(on: target, sessionDict: sessionDict)
             expect(self.error ?? false)
                 .to(beFalse(), description: "expected to fail but succeeded with result \(String(describing: result))")
             if let expectedResult = self.result {
                 expect(result).to(equal(expectedResult))
             }
+        } catch let error as CommandError {
```

Comment:
In general, we need to derive the assertions we make from the properties of the expected error, not the properties of the actual error received. As an example, we should be asserting that "errorCodeName" matches if the spec test tells us to, not based on the type of error the operations actually resulted in.

To that end, we should first focus our efforts on determining what the spec test requires in the result. To do that, we need to introduce a new `TestOperationResult` case (e.g. `.error(ErrorDescription)`). The `.document` one we're currently using is intended for successful results that happen to return a document (e.g. a successful `findOneAndDelete`).

Once we have that, we can invert the conditions and do something like:
```swift
catch {
    expect(self.error ?? false).to(beTrue(), description: "expected no error, got \(error)")
    if case let .error(errorDescription) = self.result {
        errorDescription.validateMatches(error)
    }
}
```
Feel free to rename any method / struct /  variable I've mentioned so far. I just picked them for demonstration purposes.

Reply:
I created a new `ErrorResult` struct and moved all the error checking code there. We now have a new `TestOperationResult` case `.error(ErrorResult)`.